core/src/main/java/com/linecorp/armeria/server/http/dynamic/DynamicHttpServiceBuilder.java

```diff
@@ -0,0 +1,232 @@
+    }
+
+    /**
+     * Adds a new dynamic-mapped method, invoked with given {@link HttpMethod} and (dynamically-bound) path,
+     * runs synchronously.
+     */
+    public DynamicHttpServiceBuilder run(HttpMethod method, String path, DynamicHttpFunction function,
+                                         ResponseConverter converter) {
+        return run(EnumSet.of(method), path, function, converter);
+    }
+
+    /**
+     * Adds a new dynamic-mapped method, invoked with given {@link HttpMethod}s and (dynamically-bound) path,
+     * runs synchronously.
+     */
+    public DynamicHttpServiceBuilder run(Iterable<HttpMethod> methods, String path,
+                                         DynamicHttpFunction function) {
+        DynamicHttpFunction f = DynamicHttpFunctions.of(function, converters);
+        DynamicHttpFunctionEntry entry = new DynamicHttpFunctionEntry(
+                Sets.newEnumSet(methods, HttpMethod.class), DynamicPath.of(path), f);
+        validate(entry);
+        entries.add(entry);
+        return this;
+    }
+
+    /**
+     * Adds a new dynamic-mapped method, invoked with given {@link HttpMethod}s and (dynamically-bound) path,
+     * runs synchronously.
+     */
+    public DynamicHttpServiceBuilder run(Iterable<HttpMethod> methods, String path,
+                                         DynamicHttpFunction function, ResponseConverter converter) {
+        DynamicHttpFunction composed = DynamicHttpFunctions.of(function, converter);
+        return run(methods, path, composed);
+    }
+
+    /**
+     * Add the {@link Path} annotated methods in given object.
+     *
+     * @see Path
+     * @see PathParam
+     */
+    public DynamicHttpServiceBuilder run(Object strategy) {
+        for (Method method : Methods.requestMappingMethods(strategy)) {
+            Path mapping = method.getAnnotation(Path.class);
+            EnumSet<HttpMethod> methods = Methods.httpMethods(method);
+            if (methods.isEmpty()) {
+                throw new IllegalArgumentException("HTTP Method specification is missing: " + method.getName());
+            }
+            String mappedTo = mapping.value();
+            DynamicPath dynamicPath = DynamicPath.of(mappedTo);
+            DynamicHttpFunctionImpl function = new DynamicHttpFunctionImpl(strategy, method);
+
+            Set<String> parameterNames = function.parameterNames();
+            Set<String> pathVariableNames = dynamicPath.variables();
+            if (!pathVariableNames.containsAll(parameterNames)) {
+                throw new IllegalArgumentException("Not all variables are mapped with @PathParam");
+            }
+
+            DynamicHttpFunctionEntry entry = new DynamicHttpFunctionEntry(methods, dynamicPath,
+                                                                          DynamicHttpFunctions
+                                                                                  .of(function, converters));
```

Comment:
Could look prettier this way:

    DynamicHttpFunctionEntry entry = new DynamicHttpFunctionEntry(
            methods, dynamicPath, DynamicHttpFunctions.of(function, converters));